MonacoEditorComponent/Monaco/Languages/CompletionItemProvider.cs

```diff
@@ -1,21 +1,32 @@
 ﻿using Monaco.Editor;
 using Newtonsoft.Json;
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
 using Windows.Foundation;
 
 namespace Monaco.Languages
 {
+    /// <summary>
+    /// The completion item provider interface defines the contract between extensions and
+    /// the [IntelliSense](https://code.visualstudio.com/docs/editor/intellisense)./// When computing *complete* completion items is expensive, providers can optionally implement
+    /// the `resolveCompletionItem`-function. In that case it is enough to return completion
+    /// items with a [label](#CompletionItem.label) from the
+    /// [provideCompletionItems](#CompletionItemProvider.provideCompletionItems)-function. Subsequently,
+    /// when a completion item is shown in the UI and gains focus this provider is asked to resolve
+    /// the item, like adding [doc-comment](#CompletionItem.documentation) or [details](#CompletionItem.detail).
+    /// 
+    /// </summary>
     public interface CompletionItemProvider
```

Comment:
Did only the types we were using get brought over from the generator? e.g. I don't see a new [DocumentColorProvider](https://microsoft.github.io/monaco-editor/api/interfaces/monaco.languages.documentcolorprovider.html) file, though I know we'd also need to add the corresponding register method in the LanguagesHelper manually.

Reply:
Currently I only brought types which have been implemented to monaco-editor-uwp.